uaclient/config.py

```diff
@@ -159,6 +159,12 @@ def data_path(self, key=None, private=True):
             return os.path.join(data_dir, self.data_paths[key])
         return os.path.join(data_dir, key)
 
+    def flush_cache(self):
```

Comment:
I went this direction, write_cache will now unset UAConfig locals _entitlement, _machine_token etc when written. It changed some of the logic that we relying on the stale in-memory cache, but I admin it is much cleaner to just force dirty reads instead of relying on the stale config cache.